microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyPropertyTestCase.java

```diff
@@ -0,0 +1,128 @@
+        final URL privateKeyUrl = BasicCdiTest.class.getClassLoader().getResource("pki/key.private.pkcs8.pem");
+        if (privateKeyUrl == null) {
+            throw new IllegalStateException("Private key wasn't found in resources!");
+        }
+        keyTool = RsaKeyTool.newKeyTool(privateKeyUrl.toURI());
+    }
+
+    /**
+     * A deployment with valid key. This means, that it is correct key which can be used to verify the JWT signature.
+     * @return a deployment
+     */
+    @Deployment(name = DEPLOYMENT_WITH_VALID_KEY)
+    public static WebArchive createValidKeyDeployment() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addClass(JaxRsBasicEndpoint.class)
+                .addClass(JaxRsTestApplication.class)
+                .addAsManifestResource(BasicCdiTest.class.getClassLoader().getResource("mp-config-pk-valid.properties"),
+                        "microprofile-config.properties");
+    }
+
+    /**
+     * A deployment with invalid key. The verification fails when this key is used.
+     * @return a deployment
+     */
+    @Deployment(name = DEPLOYMENT_WITH_INVALID_KEY)
+    public static WebArchive createInvalidKeyDeployment() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addClass(JaxRsBasicEndpoint.class)
+                .addClass(JaxRsTestApplication.class)
+                .addAsManifestResource(BasicCdiTest.class.getClassLoader().getResource("mp-config-pk-invalid.properties"),
+                        "microprofile-config.properties");
+    }
+
+    /**
+     * @tpTestDetails A request with proper JWT is sent on server which has configured bad public key. The server fails
+     * to verify the JWT signature and won't authorize the client. Test specification compatibility.
+     * @tpPassCrit "Unauthorized" message is shown to user and warning is present in log specifying the cause of fail.
+     * @tpSince EAP 7.3.0.CD19
+     */
+    @Test
+    @RunAsClient
+    @OperateOnDeployment(DEPLOYMENT_WITH_INVALID_KEY)
+    public void testJwkVerificationFailsWithInvalidKey(@ArquillianResource URL url) throws InterruptedException {
+        final JsonWebToken token = new JwtHelper(keyTool, "issuer").generateProperSignedJwt();
+
+        final LogListener listener = new LogListener(Pattern.compile(".*WARN.*Token is invalid: JWT rejected due to invalid signature.*"));
+        final Tailer tailer = new Tailer(WildFlyServerHelper.getPathToLogFile(), listener, 500);
+        final Thread thread = new Thread(tailer);
+        thread.start();
+
+        RestAssured.given()
+                .header("Authorization", "Bearer " + token.getRawValue())
+                .when().get(url.toExternalForm() + "basic-endpoint")
+                .then()
+                .body(equalTo("<html><head><title>Error</title></head><body>Unauthorized</body></html>"));
+
+       final boolean lineFound = Waiter.waitFor(() -> listener.getFound().get(), 10, TimeUnit.SECONDS);
```

Comment:
Isn't 10 seconds too much ? 

This timeout should just cover the tests execution, not server start period ...